assignments/array-methods.js

```diff
@@ -59,27 +59,51 @@ const runners = [
 // The event director needs both the first and last names of each runner for their running bibs. Combine both the first and last names and populate a new array called `fullNames`. This array will contain just strings.
 let fullNames = [];
 console.log(fullNames);
+runners.forEach(person => {
+  fullNames.push(`${person.first_name}  ${person.last_name}`);
+});
+console.log(fullNames);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runners' first names in uppercase because the director BECAME DRUNK WITH POWER. Populate an array called `firstNamesAllCaps`. This array will contain just strings.
 let firstNamesAllCaps = [];
 console.log(firstNamesAllCaps);
 
+firstNamesAllCaps = runners.map((fname) => fname.first_name.toUpperCase());
+console.log(firstNamesAllCaps);
+
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue. We need a filtered version of the runners array, containing only those runners with large sized shirts so they can choose a different size. This will be an array of objects.
 let runnersLargeSizeShirt = [];
 console.log(runnersLargeSizeShirt);
+// we don't want youngsters
+// use filter to produce an array with people over 21 only
+runnersLargeSizeShirt = runners.filter(largeSize => (largeSize.shirt_size === "L")); // 21 > 21 // false
+console.log(runnersLargeSizeShirt);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations and save the total into a ticketPriceTotal variable.
 let ticketPriceTotal = 0;
+ticketPriceTotal = runners.reduce((initial, person) => {
+  return initial + person.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
+
+
+
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
-
+/*
 // Problem 1
+Let us help the event director by arranging the runners alphabetically, according to their surname
```

Comment:
where are the solutions to the problems?

Reply:
I just pushed the solutions for 2 and 3. I do not thing my solution for 1 is completely functional

Reply:
Make sure all your solutions are working properly.